xds/server.go

```diff
@@ -119,10 +119,18 @@ func NewGRPCServer(opts ...grpc.ServerOption) *GRPCServer {
 	s.logger = prefixLogger(s)
 	s.logger.Infof("Created xds.GRPCServer")
 
-	creds := grpcGetServerCreds(s.gs)
-	if xc, ok := creds.(interface{ UsesXDS() bool }); ok && xc.UsesXDS() {
-		s.xdsCredsInUse = true
+	// We type assert our underlying gRPC server to the real grpc.Server here
+	// before trying to retrieve the configured credentials. This approach
+	// avoids performing the same type assertion in the grpc package which
+	// provides the implementation for internal.GetServerCredentials, and allows
+	// us to use a fake gRPC server in tests.
```

Comment:
For future consideration: moving this to a function would allow it to be stubbed for testing:
```go
var usesXDS = func(s *GRPCServer) bool { ... }
```
in tests:
```go
usesXDS = func(s *GRPCServer) bool { return s.gs.(*fakeGRPCServer).whatever }
```

Reply:
Ack